proxy/proxy.go

```diff
@@ -902,6 +903,10 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 			return nil, &proxyError{err: err}
 		}
 
+		if _, _, err := net.SplitHostPort(req.Host); err != nil {
+			req.Host = req.Host + ":" + req.URL.Port()
```

Comment:
Do you need the port in the host header?

Reply:
It is valid to put it there. Can you give some explanation about the reason as inline comment?

Reply:
The comment is showing the why, thanks @ruudk